### PR TITLE
[autoscaler] Clean up monitoring loop code

### PR DESCRIPTION
**Pull request by ericl**
## Why are these changes needed?

- Rename get_all_heartbeats to update_load_metrics
- Remove unused job pubsub handler
- Fix up the loop structure

### CODE REVIEW

**Review by wuisawesome — APPROVED** [review]
Looks pretty innocuous so long as the dead code really is dead.

**Comment by WangTaoTheTonic** [comment]
`test_multi_mode_2.py` uses monitor and it still subscribes `XRAY_JOB_PATTERN`, which should be removed.

**Comment by WangTaoTheTonic** [comment]
And also `get_all_heartbeat` is used in other places, which should be renamed.

**Review by wuisawesome — APPROVED** [review]

File: python/ray/tests/test_multi_node_2.py [hunk 1]
```diff
@@ -68,13 +68,13 @@ def setup_monitor(address):
     monitor = Monitor(
         address, None, redis_password=ray_constants.REDIS_DEFAULT_PASSWORD)
     monitor.update_raylet_map(_append_port=True)
-    monitor.psubscribe(ray.gcs_utils.XRAY_JOB_PATTERN)  # TODO: Remove?
+    monitor.subscribe(ray.ray_constants.AUTOSCALER_RESOURCE_REQUEST_CHANNEL)
```
Comment (WangTaoTheTonic) [inline python/ray/tests/test_multi_node_2.py:71]:
```suggestion
    monitor.subscribe(ray.ray_constants.AUTOSCALER_RESOURCE_REQUEST_CHANNEL)
```
We can simply remove this as no autoscaler related message should be handled in this test case.

Comment (WangTaoTheTonic) [inline python/ray/tests/test_multi_node_2.py:71]:
```suggestion
```
We can simply remove this as no autoscaler related message should be handled in this test case.

**Comment by ericl** [comment]
Ah, the reason I kept that was since an error is raised if there is no
subscriptions at all.

**Review by WangTaoTheTonic — APPROVED** [review]
LGTM :)

**Pull request merged**